api/v1alpha1/tykoasapidefinition_types.go

```diff
@@ -48,6 +52,39 @@ type TykOASReference struct {
 	ConfigmapRef ConfigMapReference `json:"configmapRef"`
 }
 
+// LocationType defines the type for the location enum
+type LocationType string
+
+// Define the allowed values for LocationType
+const (
+	LocationHeader   LocationType = "header"
+	LocationURLParam LocationType = "url-param"
+	LocationURL      LocationType = "url"
+)
+
+// TykOASVersioning contains verisoning information for an TykOASAPIDefinition.
+type TykOASVersioning struct {
+	// Default contains the default version name if a request is issued without a version.
+	Default string `json:"default"`
+
+	// Enabled is a boolean flag, if set to true it will enable versioning of the API.
+	Enabled bool `json:"enabled"`
+
+	// Key contains the name of the key to check for versioning information.
+	Key string `json:"key"`
+
+	// Location contains versioning location information. It can be one of the following:
+	// header, url-param, url.
+	// +kubebuilder:validation:Enum=header;url-param;url
```

Comment:
it'd be better if you move this validation enum to the where `LocationType` is defined above.

Reply:
Why should it be on the `LocationType` and not `Location`?

Reply:
If we add validation to `LocationType` type, whenever we use LocationType, this validation will apply. On the other, with the current approach, the validation only applies the struct field defined here.

Reply:
it's still the same. When this is converted to CRD, you get something like this. Which is what is expected.

```
      location:
        description: 'Location contains versioning location information.
          It can be one of the following: header, url-param, url.'
        enum:
        - header
        - url-param
        - url
        type: string
```

Reply:
if you think so, lets keep it as is. i'd still suggest moving all validation markers to 1 place instead of scattering them across the code.